esrally/track/params.py

```diff
@@ -679,14 +760,24 @@ def chain(*iterables):
 def create_default_reader(docs, offset, num_lines, num_docs, batch_size, bulk_size, id_conflicts, conflict_probability,
                           on_conflict, recency):
     source = Slice(io.MmapSource, offset, num_lines)
+    target = None
+    use_create = False
+    if docs.target_index:
+        target = docs.target_index
+    elif docs.target_data_stream:
+        target = docs.target_data_stream
+        use_create = True
+        if id_conflicts != IndexIdConflict.NoConflicts:
+            # can only create docs in data streams
+            raise exceptions.RallyError("Do not create readers for docs targeting data streams with conflicts")
```

Comment:
Not sure, whom is this error targetted to? If it's for a normal user, it won't help much as it uses internal terminology like "readers".

Reply:
yeh fair point its not a great message, changed to "Conflicts cannot be generated with append only data streams"